WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -1295,7 +1295,7 @@ private extension ProductFormViewController {
 private extension ProductFormViewController {
     func presentBackNavigationActionSheet(onDiscard: @escaping () -> Void = {}, onCancel: @escaping () -> Void = {}) {
         let exitForm: () -> Void = {
-            presentationStyle.createExitForm(viewController: self, completion: onDiscard)
+            presentationStyle.createExitForm(viewController: navigationController ?? self, completion: onDiscard)
```

Comment:
`ProductFormViewController` on iPhones is in a bit different view controller hierarchy.
Here you get this:

```
(lldb) po self
<_TtGC11WooCommerce25ProductFormViewControllerCS_20ProductFormViewModel_: 0x120082800>
(lldb) po self.navigationController
▿ Optional<UINavigationController>
  ▿ some : <WooCommerce.WooNavigationController: 0x10d104600>
(lldb) po self.navigationController?.viewControllers
▿ Optional<Array<UIViewController>>
  ▿ some : 1 element
    ▿ 0 : <_TtGC11WooCommerce25ProductFormViewControllerCS_20ProductFormViewModel_: 0x120082800>
```

```
(lldb) po self.navigationController?.navigationController
▿ Optional<UINavigationController>
  - some : <WooCommerce.WooTabNavigationController: 0x10d095200>
(lldb) po self.navigationController?.navigationController?.viewControllers
▿ Optional<Array<UIViewController>>
  ▿ some : 2 elements
    ▿ 0 : <WooCommerce.ProductsViewController: 0x10d105c00>
    ▿ 1 : <WooCommerce.WooNavigationController: 0x10d104600>
```